### PR TITLE
SIMSBIOHUB-395-2: Select all bulk delete sample sites

### PR DESCRIPTION
**Pull request by KjartanE**
## Links to Jira Tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-395?

## Description of Changes

- {List all relevant code changes. Include any changes to the business workflow that might not be obvious to the reviewers of this PR.}

## Testing Notes

- {List any relevant testing considerations, necessary pre-reqs, and areas of the app to focus on. Specifically, include anything that will help the reviewers of this PR verify the code is functioning as expected.}

### CODE REVIEW

File: app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx [hunk 1]
```diff
@@ -250,12 +282,43 @@ const SamplingSiteList = () => {
...
+        <Box
+          display="flex"
+          overflow="hidden"
+          alignItems="center"
+          pl={3}
+          pr={2.5}
+          height={55}
+          className="sampleSiteHeader">
+          <Checkbox
+            checked={checkboxSelectedIds.length === samplingSiteCount}
```
Comment (curtisupshall) [inline app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx:302]: I think the checkbox should be indeterminate if some (but not all) of the records are selected.

```
<Checkbox
  checked={checkboxSelectedIds.length === samplingSiteCount}
  indeterminate={checkboxSelectedIds.length > 1 && checkboxSelectedIds.length < samplingSiteCoutn}
  ...
/>

File: app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx [hunk 2]
```diff
@@ -250,12 +282,43 @@ const SamplingSiteList = () => {
...
+            onClick={() => {
+              const sampleSiteIds = sampleSites.map((sampleSite) => sampleSite.survey_sample_site_id);
+
+              if (checkboxSelectedIds.length === samplingSiteCount) {
```
Comment (curtisupshall) [inline app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx:305]: This branch should be checked before enumerating all the sample site IDs

**Review by curtisupshall — COMMENTED** [review]
Tested on remote, seems to work as intended 👍

**Review by jeznorth — COMMENTED** [review]
Approved

**Review by al-rosenthal — COMMENTED** [review]
delete works as expected.

nit: The delete button in the dialog should be replaced with a loading button, I can send multiple requests before the dialog closes causes the error snack bar to display.

File: app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx [hunk 3]
```diff
@@ -250,12 +283,43 @@ const SamplingSiteList = () => {
...
+          display="flex"
+          overflow="hidden"
+          alignItems="center"
+          pl={3}
+          pr={2.5}
+          height={55}
+          className="sampleSiteHeader">
+          <Checkbox
+            checked={checkboxSelectedIds.length === samplingSiteCount}
+            indeterminate={checkboxSelectedIds.length > 1 && checkboxSelectedIds.length < samplingSiteCount}
```
Comment (curtisupshall) [inline app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx]: Oops, should be `>= 1`

**Review by curtisupshall — APPROVED** [review]
Code looks good

**Review by NickPhura — APPROVED** [review]
🐐

**Pull request merged**